cmake/modules/SwiftModuleInstallation.cmake

```diff
@@ -18,28 +18,44 @@ function(get_swift_host_os result_var_name)
   set(${result_var_name} ${SWIFT_SYSTEM_NAME} PARENT_SCOPE)
 endfunction()
 
-function(_swift_testing_install_target module)
+# Returns the path to the Swift library installation directory
+#
+# Usage:
+#   get_swift_install_lib_dir(type result_var_name)
+#
+# Arguments:
+#   type: The type of the library (STATIC_LIBRARY, SHARED_LIBRARY, or EXECUTABLE).
+#         Typically, the value of the TYPE target property.
+#   result_var_name: The name of the variable to set
+function(get_swift_install_lib_dir type result_var_name)
```

Comment:
I kind of wonder if we should factor this out into its own module, sort of like `GNUInstallDirs`, and have it set a bunch of cached variables so that it's easier for distributors to override? Would be fine for a different PR though.

Reply:
I'm not sure if we should bother. I'd really rather just sink this down into CMake itself and support `install(TARGETS ...)` properly for Swift targets.

Reply:
Either way, the library installation directory of swift-testing is a bit different than the conventional one (not `lib/swift/<os>` but `lib/swift/<os>/testing` on darwin), so we need a project specific logic that cannot be shared across packages.